### PR TITLE
libgraphics Improvement + MultiDraw support

### PR DESCRIPTION
**Pull request by zcorniere**
Various improvement of the libgraphic, including support for the multiDrawIndirect feature

### CODE REVIEW

File: libgraphics/source/vk_utils.cxx [hunk 1]
```diff
@@ -35,6 +222,22 @@ vk::ShaderModule createShaderModule(const vk::Device &device, const std::vector<
     return device.createShaderModule(createInfo);
 }
 
+vk::SampleCountFlagBits getMexUsableSampleCount(vk::PhysicalDevice &physical_device)
```
Comment (Ersikan) [inline libgraphics/source/vk_utils.cxx:225]: `getMexUsableSampleCount` :eyes:

**Review by Ersikan — CHANGES_REQUESTED** [review]

File: libgraphics/source/Window.cxx [hunk 2]
```diff
@@ -90,7 +90,7 @@ glm::ivec2 Window::updateSize() const noexcept
 void Window::error_callback(int code, const char *msg) noexcept
 {
     logger->err("Window") << msg;
-    LOGGER_ENDL;
+
```
Comment (Ersikan) [inline libgraphics/source/Window.cxx:93]: Please run clang-format sometimes :smile:

**Review by Ersikan — COMMENTED** [review]
Definitely cleaner without all the macros.

File: libgraphics/include/pivot/graphics/AssetStorage.hxx [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+inline std::uint32_t AssetStorage::getIndex<MeshBoundingBox>(const std::string &i) const
+{
+    PIVOT_TEST_CONTAINS(meshBoundingBoxStorage, i);
+    return std::distance(meshBoundingBoxStorage.begin(), meshBoundingBoxStorage.find(i));
+}
+///@endcond
+
+#undef PIVOT_TEST_CONTAINS
+
+#endif
```
[the remark this answers is not in the document]
Comment (zcorniere) [reply libgraphics/include/pivot/graphics/AssetStorage.hxx:213]: It is inlined, so it must be implemented in the hxx

**Review by Ersikan — APPROVED** [review]

**Pull request merged**